ihatemoney/models.py

```diff
@@ -263,6 +271,37 @@ def get_pretty_bills(self, export_format="json"):
             )
         return pretty_bills
 
+    def switch_currency(self, new_currency):
+        # Update converted currency
+        if new_currency != self.default_currency:
+            if (
+                new_currency == CurrencyConverter.no_currency
+                and self.has_multiple_currencies()
+            ):
+                raise ValueError(f"Can't unset currency of project {self.id}")
+            for bill in self.get_bills_unordered():
```

Comment:
I suggest moving the `new_currency == CurrencyConverter.no_currency` test before the `for bill` loop, because the test does not depend on the bill. It's not so much about performance, but about readability.

Reply:
Do you mean either:
- Storing the result of `new_currency == CurrencyConverter.no_currency` into a variable and use it in the loop;
- or testing it and write two different loops in if/else blocks?

Reply:
I was thinking about two separate loops.

Reply:
Thanks!